File: src/m/solvers/bcgsnoneoptions.py
from pairoptions import pairoptions
from collections import OrderedDict


def bcgsmgoptions(*args):

    #retrieve options provided in *args
    options = pairoptions(*args)
    solverOptions = OrderedDict()
    solverOptions['toolkit'] = 'petsc'
    solverOptions['mat_type'] = options.getfieldvalue('mat_type', 'mpiaij')
    solverOptions['ksp_type'] = options.getfieldvalue('ksp_type', 'bcgs')
    solverOptions['pc_type'] = options.getfieldvalue('pc_type', 'none')
    return solverOptions

File: src/m/solvers/bcgsnoneoptions.m
function solverOptions=bcgsmgoptions(varargin)
%BCGSNONEOPTIONS - define PETSc solver options for the BiCGSTAB (stabilized BiConjugate Gradient) Krylov method with no preconditioning
%
%   Usage:
%      solverOptions=bcgsnoneoptions(varargin);

%retrieve options provided in varargin
options=pairoptions(varargin{:});
solverOptions=struct();
solverOptions.toolkit='petsc';
solverOptions.mat_type=getfieldvalue(options, 'mat_type','mpiaij');
solverOptions.ksp_type=getfieldvalue(options, 'ksp_type','bcgs');
solverOptions.pc_type=getfieldvalue(options, 'pc_type',  'none');
